.cloud-build/execute_changed_notebooks_helper.py

```diff
@@ -379,39 +393,30 @@ def get_changed_notebooks(
     return notebooks
 
 def _save_results(results: List[NotebookExecutionResult],
-                  accumulative_results: Dict[str,Any],
                   artifacts_bucket: str,
                   results_file: str):
 
     artifacts_bucket = artifacts_bucket.replace("gs://", "").split('/')[0]
 
-    print("Updating accumulative results ...")
+    print("Updating build results ...")
+    build_results = {}
     for result in results:
-        if result.path in accumulative_results:
-            accumulative_results[result.path]['duration'] = result.duration.total_seconds()
-            accumulative_results[result.path]['start_time'] = str(result.start_time)
-            if result.is_pass:
-                accumulative_results[result.path]['passed'] += 1
-            else:
-                accumulative_results[result.path]['failed'] += 1
-            print(f"updating {result.path}")
+        if result.is_pass:
+            pass_count = 1
+            fail_count = 0
         else:
-            if result.is_pass:
-                pass_count = 1
-                fail_count = 0
-            else:
-                pass_count = 0
-                fail_count = 1
-            accumulative_results[result.path] = {
-                    'duration': result.duration.total_seconds(),
-                    'start_time': str(result.start_time),
-                    'passed': pass_count,
-                    'failed': fail_count
-                    }
-            print(f"adding {result.path}")
+            pass_count = 0
+            fail_count = 1
+        build_results[result.path] = {
```

Comment:
The accumulation should only be done on the read path, i.e. `load_results` not in `_save_results`

Reply:
Its not accumulating. The loop is converting the NotebookExecutionResult objects into the intermediate JSON format I am saving on disk.

Reply:
You are accumulating pass count and failure count. That should be dinner done on the read path.

Reply:
looked at the code again. It is only setting pass/fail to 0 or 1 (based on result.is_pass). I don't see any accumulating. Should only record results for this build.

Reply:
Okay great, then use a bool and rename to is_pass since we don’t need an int.

Reply:
Yeah, that won't work. If I save as bool in each build results, then that breaks the code in load_results() where I accumulate the counts across plural build results. Can accumulate an int, but not a bool.

Reply:
Sure okay, let's give it a try. Thanks for hearing me out.

Reply:
It would work if you modify the read path to just count the number of True's.
The problem with the current approach is that the `pass_count` and `fail_count` is always 1, so there's no need for an int, which increases complexity (e.g. what happens another engineer comes along and changes the write path behavour, will the read-path still work?).

Reply:
I won't nitpick on this incessantly though so will defer to you.